### PR TITLE
form attachment API

### PR DESCRIPTION
**Pull request by snopoke**
@NoahCarnahan 

Simple API to get form attachments. Also add attachments field to forms returned by existing API.

### CODE REVIEW

File: corehq/apps/api/urls.py [hunk 1]
```diff
@@ -103,6 +102,7 @@ def api_url_patterns():
     for view_class in DomainAPI.__subclasses__():
         yield url(r'^custom/%s/v%s/$' % (view_class.api_name(), view_class.api_version()), view_class.as_view(), name="%s_%s" % (view_class.api_name(), view_class.api_version()))
     yield url(r'^case/attachment/(?P<case_id>[\w\-]+)/(?P<attachment_id>.*)$', CaseAttachmentAPI.as_view(), name="api_case_attachment")
+    yield url(r'^form/attachment/(?P<form_id>[\w\-]+)/(?P<attachment_id>.*)$', FormAttachmentAPI.as_view(), name="api_form_attachment")
```
Comment (dimagimon) [inline corehq/apps/api/urls.py]: E501 line too long (135 > 115 characters)

File: corehq/apps/api/object_fetch_api.py [hunk 2]
```diff
@@ -88,3 +90,22 @@ def get(self, *args, **kwargs):
...
+        if not form_id or not XFormInstance.get_db().doc_exist(form_id):
+            raise Http404
+
+        attachment_key = kwargs.get('attachment_id', None)
+
+        try:
+            resp = XFormInstance.get_db().fetch_attachment(form_id, attachment_key, stream=True)
+        except ResourceNotFound:
+            raise Http404
+
```
Comment (dimagimon) [inline corehq/apps/api/object_fetch_api.py]: W293 blank line contains whitespace

File: corehq/apps/api/urls.py (hunk 1 above)
Comment (dimagimon) [inline corehq/apps/api/urls.py]: E501 line too long (135 > 115 characters)

File: corehq/apps/api/object_fetch_api.py [hunk 3]
```diff
@@ -88,3 +90,22 @@ def get(self, *args, **kwargs):
...
```
Comment (dimagimon) [inline corehq/apps/api/object_fetch_api.py]: W293 blank line contains whitespace

**Comment by benrudolph** [comment]
this is great, the client i was working with just requested this feature

**Comment by NoahCarnahan** [comment]
:+1:

**Comment by amsagoff** [comment]
does this need documenting?

**Comment by czue** [comment]
I think @snopoke already documented as per commcare-developers thread

**Comment by benrudolph** [comment]
i don't think either of them (case and form) are on the confluence page though. i haven't heard of commcare-developers thread, assuming that's the developer's equivalent of commcare-users

**Comment by czue** [comment]
oh, good point, looks like @snopoke just put the docs in the thread: https://groups.google.com/forum/#!topic/commcare-developers/EuugDu1h--A

we should probably add to confluence

**Comment by snopoke** [comment]
Yup, will add to confluence.

**Pull request merged**